examples/cpp/pyperf/PyPerfBPFProgram.cc

```diff
@@ -473,14 +475,24 @@ get_first_arg_name(
   char *argname,
   size_t maxlen) {
   int result = 0;
+  ssize_t ob_size; // Py_ssize_t;
   // Roughly equivalnt to the following in GDB:
   //
   //   ((PyTupleObject*)$frame->f_code->co_varnames)->ob_item[0]
   //
   void* args_ptr;
   result |= bpf_probe_read_user(&args_ptr, sizeof(void*), code_ptr + offsets->PyCodeObject.co_varnames);
-  result |= bpf_probe_read_user(&args_ptr, sizeof(void*), args_ptr + offsets->PyTupleObject.ob_item);
-  result |= bpf_probe_read_user_str(argname, maxlen, args_ptr + offsets->String.data);
+  result |= bpf_probe_read_user(&ob_size, sizeof(ob_size), args_ptr + offsets->String.size); // String.size is PyVarObject.ob_size
+  if (ob_size > 0) {
```

Comment:
This shouldn't be in this function. It's better if it'd be in `get_classname` such that when this condition is false we skip the `first_self` and `first_cls` tests.

Reply:
It kinda breaks up that function, then. (because now I will read `co_varnames` outside). So I can remove it altogether and inline it into `get_classname`.

Otoh, this is only to avoid 4 int comparisons? I think the API is legit this way.

Reply:
5 comparisons and 4 stores, I think.

Reply:
Eh, this is not relevant in runtime. I think that merging the 2 functions will produce a less concise API